plugin/kubernetes/README.md

```diff
@@ -101,6 +101,20 @@ kubernetes [ZONES...] {
 
 Enabling zone transfer is done by using the *transfer* plugin.
 
+## Startup
+
+When CoreDNS starts with the *kubernetes* plugin enabled, it will delay serving DNS for up to 5 seconds
+until it can connect to the Kubernetes API and synchronize all object watches.  If this cannot happen within
+5 seconds, then CoreDNS will start serving DNS while the *kubernetes* plugin continues to try to connect
+and synchronize all object watches.  CoreDNS will answer SERVFAIL to any request made for a Kubernetes record
+that has not yet been synchronized.
```

Comment:
This seems like a bad idea. CoreDNS should at least report unready until the kubernetes plugin is ready.  Do I understand correctly that CoreDNS could start, report ready to health probes, and respond with SERVFAIL to DNS queries? What do you think about shortcircuiting this logic, or increasing the timeout?

Actually though, doesn't https://github.com/openshift/coredns/blob/20f79176b920390fbb12c4a87aa190969d0e59dc/plugin/kubernetes/ready.go#L4 prevent prevent CoreDNS from reporting ready?  Maybe this paragraph is just incomplete and misleading.

Reply:
I think this paragraph fails to mention issues about Readyness for the kubernetes plugin.

I think you are correct in that the kubernetes plugin wouldn't report ready until objects have been syncrhonized, so I don't think we need any _code_ changes (downstream or upstream).

Shall we file an issue upstream to point out these confusing docs?

Reply:
Yeah, an issue or PR to clarify that the ready plugin does the right thing would be useful.

Reply:
Ah, looks like the `Ready` section of the [kubernetes plugin readme](https://github.com/coredns/coredns/blob/master/plugin/kubernetes/README.md) answers our question:

```
Ready
This plugin reports readiness to the ready plugin. This will happen after it has synced to the Kubernetes API
```

So I think we don't need to make any changes upstream here.